sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.sql.types._
 object NestedColumnAliasing {
 
   def unapply(plan: LogicalPlan)
-    : Option[(Map[GetStructField, Alias], Map[ExprId, Seq[Alias]])] = plan match {
+    : Option[(Map[ExtractValue, Alias], Map[ExprId, Seq[Alias]])] = plan match {
     case Project(projectList, child)
         if SQLConf.get.nestedSchemaPruningEnabled && canProjectPushThrough(child) =>
```

Comment:
During reviews, shall we test with `spark.sql.optimizer.nestedSchemaPruning.enabled=true`?

Reply:
`NestedColumnAliasingSuite` enables `spark.sql.optimizer.nestedSchemaPruning.enabled`. Is this you asked?

Reply:
No, I want to run for all Spark UT~

Reply:
Oh, I see. Then I will change the default value to `true` and see if all tests can pass. Before merging, we can revert it back.

Reply:
Yes~ Thanks!